draft-ietf-quic-transport.md

```diff
@@ -1475,6 +1475,49 @@ lifetime of a connection, especially in response to connection migration
 ({{migration}}); see {{issue-cid}} for details.
 
 
+## Authenticating Connection IDs {#cid-auth}
+
+The choice each endpoint makes about connection IDs during the handshake is
+authenticated by including all values in transport parameters; see
+{{transport-parameters}}. This ensures that all connection IDs used for the
+handshake are also authenticated by the cryptographic handshake.
+
+Each endpoint includes the value of the Source Connection ID field from the most
+recent Initial packet it sent in the initial_connection_id transport
+parameter; see {{transport-parameter-definitions}}. A server includes the
+Destination Connection ID field it receives in original Initial packets from the
+client - Initial packets received by the server prior to sending a Retry packet
+- in the original_connection_id transport parameter. After sending a Retry
+packet, a server also includes the Source Connection ID field from the Retry
+packet in the retry_connection_id transport parameter.
+
+The values provided by a peer for these transport parameters MUST match the
+values that an endpoint used in the Destination Connection ID field of Initial
+packets that it sent. Including connection ID values in transport parameters and
+verifying them ensures that that an attacker cannot influence the choice of
+connection ID for a successful connection by injecting packets carrying
+attacker-chosen connection IDs during the handshake. An endpoint MUST
+treat any of the following as a connection error of type PROTOCOL_VIOLATION:
+
+* absence of the initial_connection_id transport parameter from either
+  endpoint,
+
+* absence of the original_connection_id transport parameter from the server,
+
+* absence of the retry_connection_id transport parameter from the server after
+  receiving a Retry packet,
+
+* presence of the retry_connection_id transport parameter when no Retry packet
+  was received, or
+
+* a mismatch between values received from a peer in these transport parameters
+  and the value sent in the corresponding Destination Connection ID fields of
+  Initial packets.
```

Comment:
```suggestion
  Initial packets or the Source Connection ID field of a Retry packet.
```
...though perhaps this is fine, since the client will then use this value as its DCID for the next Initial packet.

Reply:
> the client will then use this value as its DCID for the next Initial

Yeah, that was my thought.  The Retry.SCID thing is a choice people are free to use, but that doesn't need to be written down.  This way is simpler.